am-lib/src/test/java/uk/gov/hmcts/reform/amlib/PermissionTest.java

```diff
@@ -0,0 +1,136 @@
+    }
+
+    @Test
+    public void fromSumOf_sumOfPermissionsValueSeven_ExpectArray() throws UnsupportedPermissionsException {
+        Set<Permission> permissions = Permissions.fromSumOf(7);
+
+        assertThat(permissions).containsOnly(Permission.CREATE, Permission.READ, Permission.UPDATE);
+    }
+
+    @Test
+    public void fromSumOf_sumOfPermissionsValueTen_ExpectArray() throws UnsupportedPermissionsException {
+        Set<Permission> permissions = Permissions.fromSumOf(10);
+
+        assertThat(permissions).containsOnly(Permission.SHARE, Permission.READ);
+    }
+
+    @Test
+    public void fromSumOf_sumOfPermissionsValueThirteen_ExpectArray() throws UnsupportedPermissionsException {
+        Set<Permission> permissions = Permissions.fromSumOf(13);
+
+        assertThat(permissions).containsOnly(Permission.CREATE, Permission.UPDATE, Permission.SHARE);
+    }
+
+    @Test
+    public void fromSumOf_sumOfPermissionsValueFifteen_ExpectArray() throws UnsupportedPermissionsException {
+        Set<Permission> permissions = Permissions.fromSumOf(15);
+
+        assertThat(permissions).containsOnly(Permission.CREATE, Permission.READ, Permission.UPDATE, Permission.SHARE);
+    }
+
+    @Test
+    public void fromSumOf_sumOfPermissionsValueSeventeen_ExpectArray() throws UnsupportedPermissionsException {
+        Set<Permission> permissions = Permissions.fromSumOf(17);
+
+        assertThat(permissions).containsOnly(Permission.CREATE, Permission.DELETE);
+    }
+
+    @Test
+    public void fromSumOf_sumOfPermissionsValueTwentyTwo_ExpectArray() throws UnsupportedPermissionsException {
+        Set<Permission> permissions = Permissions.fromSumOf(22);
+
+        assertThat(permissions).containsOnly(Permission.READ, Permission.DELETE, Permission.UPDATE);
+    }
+
+
+    @Test
+    public void fromSumOf_sumOfPermissionsValueTwentyFive_ExpectArray() throws UnsupportedPermissionsException {
+        Set<Permission> permissions = Permissions.fromSumOf(25);
+
+        assertThat(permissions).containsOnly(Permission.CREATE, Permission.SHARE, Permission.DELETE);
+    }
+
+    @Test
+    public void fromSumOf_sumOfPermissionsValueTwentyEight_ExpectArray() throws UnsupportedPermissionsException {
+        Set<Permission> permissions = Permissions.fromSumOf(28);
+
+        assertThat(permissions).containsOnly(Permission.SHARE, Permission.DELETE, Permission.UPDATE);
+    }
+
+    @Test
+    public void fromSumOf_sumOfPermissionsValueThirtyOne_ExpectArray() throws UnsupportedPermissionsException {
+        Set<Permission> permissions = Permissions.fromSumOf(31);
+
+        assertThat(permissions).containsOnly(
+            Permission.CREATE, Permission.READ, Permission.UPDATE, Permission.SHARE, Permission.DELETE);
+    }
+
+    @Test
+    public void fromSumOf_sumOfPermissionsValueHighValue_ShouldThrowUnsupportedPermission() {
+        assertThatExceptionOfType(UnsupportedPermissionsException.class).isThrownBy(
+            () -> Permissions.fromSumOf(32))
```

Comment:
I'd sum up all values that we use and then increment it by 1.

Reply:
👍